core/src/main/java/com/google/cloud/sql/core/SslSocketFactory.java

```diff
@@ -615,21 +645,23 @@ public long getLastFailureMillis() {
   }
 
   private static class InstanceSslInfo {
-    private final String instanceIpAddress;
+    private final HashMap<String, String> instanceIpAddresses = new HashMap<>();
     private final X509Certificate ephemeralCertificate;
     private final SSLSocketFactory sslSocketFactory;
 
     InstanceSslInfo(
-        String instanceIpAddress,
         X509Certificate ephemeralCertificate,
         SSLSocketFactory sslSocketFactory) {
-      this.instanceIpAddress = instanceIpAddress;
       this.ephemeralCertificate = ephemeralCertificate;
       this.sslSocketFactory = sslSocketFactory;
     }
 
-    public String getInstanceIpAddress() {
-      return instanceIpAddress;
+    public void setInstanceIpAddress(String type, String ipAddress) {
```

Comment:
What is this supposed to be doing? I don't think set is the correct naming choice here, since it's really `put`ing the IP, which is sometimes adding a new object and sometimes replacing one. Can a type be replaced?

Reply:
I renamed it to put for now. There should be at most one ip per type, but I don't want the code to panic if that is ever not true.